spec/controllers/api/v1/messages_controller_spec.rb

```diff
@@ -28,20 +32,55 @@
       expect(subject['messages'].length).to eq(2)
     end
 
-    describe do
+    describe 'filtering messages' do
       before { 2.times { create :message } }
 
-      it "for item" do
+      it "for one item" do
         3.times { create :message, item: item }
         get :index, item_id: item.id
         expect(subject['messages'].length).to eq(3)
       end
 
-      it "for offer" do
+      it "for multiple items" do
+        3.times { create :message, item: item }
+        3.times { create :message, item: item2 }
+        get :index, item_id: "#{item.id},#{item2.id}"
+        expect(subject['messages'].length).to eq(6)
+      end
+
+      it "for one offer" do
         3.times { create :message, offer: offer }
         get :index, offer_id: offer.id
         expect(subject['messages'].length).to eq(3)
       end
+
+      it "for multiple offers" do
+        3.times { create :message, offer: offer }
+        3.times { create :message, offer: offer2 }
+        get :index, offer_id: "#{offer.id},#{offer2.id}"
+        expect(subject['messages'].length).to eq(6)
+      end
+
+      it "for one order" do
+        3.times { create :message, order: order }
+        get :index, order_id: order.id
+        expect(subject['messages'].length).to eq(3)
+      end
+
+      it "for multiple orders" do
+        3.times { create :message, order: order }
+        3.times { create :message, order: order2 }
+        get :index, order_id: "#{order.id},#{order2.id}"
+        expect(subject['messages'].length).to eq(6)
+      end
+
+      it "for a certain state" do
+        3.times { create :message, offer: offer }
+        3.times { create :message, offer: offer, sender_id: user.id }
+        3.times { create :message, offer: offer2 }
```

Comment:
Which messages have read status here?

Reply:
the ones created by the user are marked as read by default